hashicorp/dev-portal
# Developer AI

# Description

This introduces "Developer AI". 
This work comes in largely 3 chunks:
1. The new chatbox UI & the streaming processing logic/state handling
2. The new chat server endpoints
3. Moving around existing code to make the UI _fit_

<img width="932" alt="CleanShot 2023-09-08 at 10 13 05@2x" src="https://github.com/hashicorp/dev-portal/assets/26389321/9fc2a0fc-1c03-42a9-9885-0c371b8418be">

### To test

> [!IMPORTANT]
> To gain access to this feature, you need to complete the following steps:

- You need to be "opted in". To opt-in:
   - log in to https://dev-portal-git-kwai-hashicorp.vercel.app/profile/account
   - grab your "HashiCorp ID" and send it to @thiskevinwang 
- Once opted in, you should see a Toast pop up, notifying you how/where to discover the feature.

### Checklist
- [x] Ensure production has `EXPERIMENTAL_CHAT_API_BASE_URL` env var

---
Builds on work from
- https://github.com/hashicorp/dev-portal/pull/2148
- https://github.com/hashicorp/dev-portal/pull/2151
- #2155
- https://github.com/hashicorp/dev-portal/pull/2157
- https://github.com/hashicorp/dev-portal/pull/2161
- https://github.com/hashicorp/dev-portal/pull/2163
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204865849100145
